rebasehelper/helpers/lookaside_cache_helper.py

```diff
@@ -167,24 +167,33 @@ def post(check_only=False):
             cd = ChunkedData(check_only)
             r = requests.post(url, data=cd, headers=cd.headers, auth=auth)
             if not 200 <= r.status_code < 300:
-                raise LookasideCacheError(r.reason)
+                raise LookasideCacheError('{0}: {1}'.format(r.reason, r.text.strip()))
             return r.content
 
         path = os.path.join(source_dir, filename)
-        state = post(check_only=True)
+
+        try:
+            state = post(check_only=True)
+        except (requests.exceptions.ConnectionError, LookasideCacheError) as e:
+            # just log the error and bail out
+            logger.error('Attempt to upload to lookaside cache failed: %s', str(e))
+            return
+
         if state.strip() == b'Available':
             # already uploaded
+            logger.info('Source is already present in lookaside cache')
             return
 
         logger.info('Uploading %s to lookaside cache', path)
         try:
-            post()
-        except requests.exceptions.ConnectionError as e:
+            try:
```

Comment:
Is this inner try-finally block necessary? Shouldn't it be sufficient to keep the finally part connected to the outer `try-except-finally` block?